core/types/calls.cc

```diff
@@ -1995,21 +1995,27 @@ class Class_new : public IntrinsicMethod {
     }
 
     void apply(const GlobalState &gs, const DispatchArgs &args, DispatchResult &res) const override {
-        auto mustExist = true;
-        ClassOrModuleRef self = unwrapSymbol(gs, args.thisType, mustExist);
-
-        auto attachedClass = self.data(gs)->attachedClass(gs);
-        if (!attachedClass.exists()) {
-            // If someone takes `klass: T::Class[T.anything]` and calls `klass.new`, the call is
-            // actually going to be on an "instance" not a singleton (Class.new, the one on the
-            // singleton, is the one that defines a new class at runtime).
-            //
-            // In that case, there's no attachedClass to look for an `initialize` method on.
-            // We could _maybe_ imagine trying to dispatch to `initialize` on the `<AttachedClass>`
-            // type argument? But I haven't thought about what the consequences of that would be.
```

Comment:
I've now thought through the consequences, and I think it's the right thing to do 🙂